### PR TITLE
refactor: refactoring store and options type

### PR DESCRIPTION
**Pull request by ssi02014**
## Summary
While looking at the code, I noticed that the `Store` type was defined repeatedly. I thought it would be better if I exported and utilized the Store from `store.ts`.

I also improved the `Options` type.
It's a simple improvement suggestion. I would appreciate it if you could review it. 

Thanks for providing such a great library. 🙏

## Check List

- [x] `yarn run prettier` for formatting code and docs

### CODE REVIEW

File: src/vanilla/store.ts [hunk 1]
```diff
@@ -795,7 +795,7 @@ export const createStore = () => {
   }
 }
 
-type Store = ReturnType<typeof createStore>
+export type Store = ReturnType<typeof createStore>
```
Comment (dai-shi) [inline src/vanilla/store.ts:798]: While it's controversial, this is intentional not to export the type here. We basically strictly control what type to export.

File: src/react/useAtomValue.ts [hunk 2]
```diff
@@ -41,8 +39,7 @@ const use =
     }
   })
 
-type Options = {
-  store?: Store
+type Options = Parameters<typeof useStore>[0] & {
```
Comment (dai-shi) [inline src/react/useAtomValue.ts:42]: This looks good.

**Review by dai-shi — CHANGES_REQUESTED** [review]
Can you revert a part of change?

File: src/vanilla/store.ts (hunk 1 above)
Comment (ssi02014) [reply src/vanilla/store.ts]: @dai-shi Thanks for the quick feedback 🤗 I have reverted some of the work.

**Comment by dai-shi** [comment]
**Size Change:** -8 B (0%) 

**Total Size:** 70.3 kB

| Filename | Size | Change |
| :--- | :---: | :---: |
| `dist/system/vanilla.development.js` | 4.16 kB | -3 B (0%) |
| `dist/system/vanilla.production.js` | 2.08 kB | -2 B (0%) |
| `dist/umd/vanilla.development.js` | 4.94 kB | -1 B (0%) |
| `dist/umd/vanilla.production.js` | 2.25 kB | -4 B (0%) |
| `dist/vanilla.js` | 4.86 kB | +2 B (0%) |

<details><summary>ℹ️ <strong>View Unchanged</strong></summary>

| Filename | Size |
| :--- | :---: |
| `dist/babel/plugin-debug-label.js` | 907 B |
| `dist/babel/plugin-react-refresh.js` | 1.14 kB |
| `dist/babel/preset.js` | 1.38 kB |
| `dist/index.js` | 242 B |
| `dist/react.js` | 1.06 kB |
| `dist/react/utils.js` | 1.26 kB |
| `dist/system/babel/plugin-debug-label.development.js` | 1.09 kB |
| `dist/system/babel/plugin-debug-label.production.js` | 757 B |
| `dist/system/babel/plugin-react-refresh.development.js` | 1.29 kB |
| `dist/system/babel/plugin-react-refresh.production.js` | 929 B |
| `dist/system/babel/preset.development.js` | 1.56 kB |
| `dist/system/babel/preset.production.js` | 1.13 kB |
| `dist/system/index.development.js` | 253 B |
| `dist/system/index.production.js` | 182 B |
| `dist/system/react.development.js` | 1.18 kB |
| `dist/system/react.production.js` | 711 B |
| `dist/system/react/utils.development.js` | 722 B |
| `dist/system/react/utils.production.js` | 461 B |
| `dist/system/utils.development.js` | 258 B |
| `dist/system/utils.production.js` | 188 B |
| `dist/system/vanilla/utils.development.js` | 4.73 kB |
| `dist/system/vanilla/utils.production.js` | 2.88 kB |
| `dist/umd/babel/plugin-debug-label.development.js` | 1.06 kB |
| `dist/umd/babel/plugin-debug-label.production.js` | 845 B |
| `dist/umd/babel/plugin-react-refresh.development.js` | 1.28 kB |
| `dist/umd/babel/plugin-react-refresh.production.js` | 1.01 kB |
| `dist/umd/babel/preset.development.js` | 1.53 kB |
| `dist/umd/babel/preset.production.js` | 1.24 kB |
| `dist/umd/index.development.js` | 381 B |
| `dist/umd/index.production.js` | 329 B |
| `dist/umd/react.development.js` | 1.18 kB |
| `dist/umd/react.production.js` | 785 B |
| `dist/umd/react/utils.development.js` | 1.43 kB |
| `dist/umd/react/utils.production.js` | 1.03 kB |
| `dist/umd/utils.development.js` | 398 B |
| `dist/umd/utils.production.js` | 343 B |
| `dist/umd/vanilla/utils.development.js` | 5.64 kB |
| `dist/umd/vanilla/utils.production.js` | 3.47 kB |
| `dist/utils.js` | 246 B |
| `dist/vanilla/utils.js` | 5.5 kB |

</details>



<a href="https://github.com/preactjs/compressed-size-action"><sub>compressed-size-action</sub></a>

File: src/vanilla/store.ts [hunk 3]
```diff
@@ -176,7 +176,7 @@ export const createStore = () => {
     if (import.meta.env?.MODE !== 'production') {
       Object.freeze(atomState)
     }
-    const prevAtomState = atomStateMap.get(atom)
+    const prevAtomState = getAtomState(atom)
```
Comment (dai-shi) [inline src/vanilla/store.ts:179]: Good catch.

**Review by dai-shi — APPROVED** [review]
LGTM! Thanks for your contribution.

I will merge this when I prepare the next release.

**Comment by ssi02014** [comment]
@dai-shi 
Thanks for the quick review, and I'm glad to contribute to such a great library 🚀

**Pull request merged**